### ISSUE TITLE
V fmt seems to convert union into struct ("Struct Of The Union" ?)

### ISSUE THREAD

**Issue by nsauzede** [vlang/v#6718]
OS: linux, Ubuntu 20.10 (VM)
Processor: 2 cpus, 64bit, little endian, Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
CC version: cc (Ubuntu 10.2.0-13ubuntu1) 10.2.0

vroot: /home/nsauzede/perso/git/v
vexe: /home/nsauzede/perso/git/v/v
vexe mtime: 2020-11-02 11:14:41
is vroot writable: true
V full version: V 0.1.29 4e760c7

Git version: git version 2.27.0
Git vroot status: latest-commit-159-g4e760c70 (1 commit(s) behind V master)
.git/config present: true
/var/tmp/tcc status: master 5e73982e
thirdparty/tcc: N/A

**What did you do?**
I tried to format this simple source :
```go
union Un {
        i int
        b byte
}

fn main() {
        mut u := Un{
                i: 1234
        }
        println('u=$u')
        u = Un{
                b: 42
        }
        println('u=$u')
}
```

**What did you expect to see?**
I expected the union to keep on being an union, and not turned into a struct
 
**What did you see instead?**
```go
struct Un {
        i int
        b byte
}

fn main() {
        mut u := Un{
                i: 1234
        }
        println('u=$u')
        u = Un{
                b: 42
        }
        println('u=$u')
}
```
Notice the union that is turned to a struct

**Comment by spytheman** [comment]
Fixed in ae24178

**Comment by nsauzede** [comment]
I confirm it works ! Thanks @spytheman !!
